Source/Editor/CustomEditors/Editors/SpriteHandleEditor.cs

```diff
@@ -54,10 +57,20 @@ public override void Initialize(LayoutElementsContainer layout)
             var spritesCount = value.SpritesCount;
             var spritePicker = spriteLabel.ComboBox();
             spritePicker.ComboBox.Items.Capacity = spritesCount;
-            for (int i = 0; i < spritesCount; i++)
-            {
-                spritePicker.ComboBox.AddItem(value.GetSprite(i).Name);
+            _guids = new Guid[spritesCount];
+
+            var sortedSprites = value.Sprites.Values.ToList();
+            sortedSprites.Sort((s0, s1) => {
+                var xComp = s0.Area.Location.X.CompareTo(s1.Area.Location.X);
+
+                return xComp == 0 ? s0.Area.Location.Y.CompareTo(s1.Area.Location.Y) : xComp;
```

Comment:
This sorting is more or less duplicated here and in the `SpriteAtlasWindow`, but I have a PR in the works that completely replaces this editor anyway, so I decided not to spend a lot of time in refactors for now.